tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/DependencyParsingTransitionSystem.scala

```diff
@@ -83,3 +93,83 @@ object DependencyParsingTransitionSystem {
   val keywords = WordClusters.commonWords ++ WordClusters.puncWords ++
     WordClusters.stopWords
 }
+
+/** The LabelLeftArc operator labels the most recently created left-facing arc. */
+case class LabelLeftArc(val label: Symbol) extends TransitionParserStateTransition {
+
+  override def satisfiesPreconditions(state: TransitionParserState): Boolean = {
+    state.parserMode == DependencyParserModes.LEFTLABEL
+  }
+
+  override def advanceState(state: TransitionParserState): State = {
+    require(state.previousLink != None, s"Cannot proceed without an arc to label: $state")
+    state.previousLink match {
+      case Some((crumb, gretel)) =>
+        if (PolytreeParse.arcInverterStanford.inverseArcLabels.contains(label)) {
+          val gretelChildren: Set[Int] =
+            state.children.getOrElse(gretel, Set.empty[Int])
+          state.copy(
+            arcLabels = state.arcLabels +
+            (Set(crumb, gretel) -> label),
+            parserMode = DependencyParserModes.TRANSITION,
+            children = state.children +
+            (gretel -> (gretelChildren + crumb))
+          )
+        } else {
+          val crumbChildren: Set[Int] =
+            state.children.getOrElse(crumb, Set.empty[Int])
+          state.copy(
+            arcLabels = state.arcLabels +
+            (Set(crumb, gretel) -> label),
+            parserMode = DependencyParserModes.TRANSITION,
+            children = state.children +
+            (crumb -> (crumbChildren + gretel))
+          )
+        }
+      case _ => ???
```

Comment:
If you're doing this anyways, with the `require` at the top and everything, why not do this:
```
val (crumb, gretel) = state.previousLink.get
```

You get rid of two levels of nesting, and don't have to use `???`.